designs/virtualization/4.6/Next-run-configuration/Next-run-configuration.md

```diff
@@ -0,0 +1,62 @@
+---
+parent: Virtualization
+version: 4.6
+---
+
+# Editing
+
+When a user clicks to edit/ update one of the attributes (f.e. Flavor) it will open a modal with an alert notification letting the user know that the VM must be restarted in order to see the changes applied. The alert will persist until they restart the VM.
+Alternatively they could clear the changes and keep what they had.
+To avoid confusion they will be able to view the current settings in the VM Details page.
+
+![user is editing values](img/EditOriginal.png)
+
+At this point the user can either just save the requested changes and postpone the restart or save and restart the VM.
+
+## Restart modal
+
+If the user chooses to save and restart the VM they will be asked again to ensure that and will get a modal including an inline notification that restarting may be disruptive to some logged in users and a notification that the new setting will be applied upon restart. The notification will include a confirmation checkbox so they would be aware of the effect that the restart action might cause and a link to the logged in users list at the bottom of the VM Details page.
+
+![user chooses to resart the VM](img/RestartModal.png)
+
+## VM Details page
+
+In the VM Details page they will be notified that the changes they have made will get applied only after the VM will be restarted. 
+Next to the ‘Flavor’ section we will add a link that will lead to the Flavor modal where they can view the pending changes.
+Only in cases when users have made changes to the part of the YAML which are not exposed in the ui, we will add a link to ‘View pending changes in YAML editor’.
+
+![VM details page](img/vm-details.png)
+
+In cases users made changes to the part of the YAML which are not exposed in the ui we’ll add the option to ‘View pending changes in the YAML editor.
+
+![Users can view the changes in the YAML](img/ViewYAML-editor.png)
+
+## List view
+
+In the List view, we’ll show the Running* status. Clicking it will open a popover explaining that this VM has some pending changes that will apply once it will be restarted.
+Users will be able to restart from here to save the step of clicking into the kebab menu.
+
+![Popover indicating a restart is required from the list view](img/ListView.png)
+
+## Adding a ‘Clear changes’ option
+
+Users will be able to view the pending changes from the ‘Edit’ modals and also have the option to clear the changes they have made by clicking the ‘Clear changes’ link.
+
+![User edit and makes changes to values](img/Edit.png)
+
+If they will choose to clear changes - that action will remove the changes they have made and the current settings will stay the same.
+
+![Clear changes modal](img/ClearChanges2.png)
```

Comment:
What does this modal look like if the user clicks the "Clear changes" inline alert on the Details page that could potentially list multiple pending changes for Flavor, Devices, etc.? Should each of those be listed in the modal's text here like this mockup does for flavor? Or would the modal not list them all and just be more general?

Reply:
In this case, I think the modal would include a list of all the pending changes with checkboxes so the user can select which ones they want to clear and which not.
WDYT? Can we support multiple selections now?

Reply:
@matthewcarleton what if they want to clear only 1 change out of many?

Reply:
I'd say yes as a first pass on this. If we get feedback that it's really frustrating that they have to go to each one we could expand on it. I think we have to remember this is not a common thing a user will want to do. We shouldn't see a lot of cases where the user has made changes -> saved them -> and then regrets them. We just want to ensure they are covered if they find themselves in this situation.

Reply:
A simple hack to clear some changes is to change them back to their previous configuration